Database/CleanArchive.py

```diff
@@ -88,12 +96,69 @@ def clean(all=True):
         transaction = {
             '$pull': {'snippets': {'transcript': ''}}
         }
+        if duplicate < 1.0:
+            print('Duplicate found')
+            set_fields['duplicate'] = True
+            duplicates['metadata.Subtitle'] += 1
+        else:
+            set_fields['duplicate'] = False
         if errors:
             set_fields['errors'] = errors
         else:
             transaction['$unset'] = {'errors': 1}
         if set_fields:  # $set cannot be empty.
             transaction['$set'] = set_fields
+
+
+
         db.ArchiveIndex.update_one({'_id': episode['_id']}, transaction)
     print('Updates:', updates)
     print('Failures:', failures)
+    print('Duplicates', duplicates)
+
+
+def checkDuplicate(epi):
+    upperBound = db.ArchiveIndex.find({'identifier': epi['identifier']})[0]['date']
+    dateTimeObj = datetime.datetime.strptime(upperBound, '%Y-%m-%dT%H:%M:%SZ')
+    lowerBound = str(dateTimeObj - dateTimeObj - datetime.timedelta(days=1))
+    fields = db.ArchiveIndex.find({'date': {'$lte': upperBound, '$gte': lowerBound}}, {'snippets': 1})
+    currentEpisode = ''
+    for snip in epi['snippets']:
+        currentEpisode += str(snip['transcript'])
+    for field in fields:
+        compareEpisode = ''
+        cosine = 0
+        try:
+            for snip in field['snippets']:
+                compareEpisode += str(snip['transcript'])
+            xList = word_tokenize(currentEpisode)
+            yList = word_tokenize(compareEpisode)
+            sw = stopwords.words('english')
+            l1 = []
+            l2 = []
+            xSet = {w for w in xList if not w in sw}
+            ySet = {w for w in yList if not w in sw}
+            vector = xSet.union(ySet)
+            for w in vector:
+                if (w in xSet):
+                    l1.append(1)
+                else:
+                    l1.append(0)
+                if (w in ySet):
+                    l2.append(1)
+                else:
+                    l2.append(0)
+                c = 0
+            for i in range(len(vector)):
+                c += l1[i] * l2[i]
+            cosine = c / float((sum(l1) * sum(l2)) ** .5)
+            # print(cosine)
+
+
+        except Exception as e:
+            print('There was an issue comparing two of the transcripts')
+        if cosine > 0.5:
+            print(cosine)
+            return cosine
+
+    return 1
```

Comment:
Why return 1? Won't this mark the episode as a duplicate?

Reply:
I originally set it to return 0 but it was always returning 1 so I set it to that anyway. 
if duplicate < 1.0:
            print('Duplicate found')
            errors.append('transcript_is_duplicate')
            duplicates['metadata.Title'] += 1

Reply:
If it didn't return 0, then it never reached that line.
I'm not a huge fan of that return type. Not sure a better alternative yet though. I'll think about it.